MVPCompetitionTask/Pages/ManageListingPages/DeleteShareSkillPage.cs

```diff
@@ -1,61 +1,101 @@
+    //Collections to store element collection
     private ReadOnlyCollection<IWebElement>? rowElements, colElements, elements;
+
     private string shareSkillTitle;
     private ExtentReports testReport;
     private ExtentTest test;
     private bool rowExists;
+    
+
+    //Element repository for POM design pattern
+    private readonly By manageListings = By.XPath("//a[text()='Manage Listings']");
+    private readonly By manageListingsRows = By.XPath("//tr");
+    private readonly By manageListingsColumns = By.TagName("td");
+    private readonly By manageListingsPageBtns = By.TagName("button");
+
+    private readonly CommonSendKeysAndClickElements elementInteractions;
+
     public DeleteShareSkillPage()
     {
+        elementInteractions = new CommonSendKeysAndClickElements();
         rowExists = false;
         shareSkillTitle = "";
-        findElements = new CommonSendKeysAndClickElements();
         testReport = IExtentRpt.testReport;
         test = testReport.CreateTest("Test_DeleteShareSkill" + DateTime.Now.ToString("_hhmmss")).Info("Deleting A Share Skill");
+    }
+
+    public void ClickOnManageListings()
+    {
+        //Click on manage listings button
+        elementInteractions.ClickElement(manageListings);
+    }
 
+    public void GetShareSkillRows()
+    {
+        //Getting all share skill rows
+        rowElements = elementInteractions.ReturnElementCollectionByPresenceOfAllElements(manageListingsRows);
+    }
+
+    public void GetShareSkillRowColumns(int rowID)
+    {
+        //Getting all the column elements in each row in manage listings
+        colElements = rowElements[rowID].FindElements(manageListingsColumns);
+    }
+
+    public void ClickOnDeleteShareSkillBtn()
+    {
+        //clicking the delete button from the 8th column of the row
+        colElements[7].FindElements(manageListingsPageBtns)[2].Click();
+    }
+
+    public void ConfirmDeleteInAlertWindow()
+    {
+        //Confirming delete
+        elements = elementInteractions.ReturnElementCollectionByElementISVisible(manageListingsPageBtns);
+        elements[elements.Count - 1].Click();
     }
 
     public void DeleteShareSkill(string shareSkillTitle)
     {
         this.shareSkillTitle = shareSkillTitle;
-        findElements.clickOnElement(nameof(By.XPath), "//a[text()='Manage Listings']");
-        //Getting all the row elements for manage listings
-        rowElements = findElements.findElementsByLocator(nameof(By.XPath), "//tr");
+        ClickOnManageListings();
+        GetShareSkillRows();
+        //Checking if a valid share skill is entered for delete operation
+        if (CheckShareSkill() == false)
+            rowExists.Should().BeTrue("Share Skill Not Found....Enter a valid share skill to delete");
+        test.Log(Status.Info, "Share skill deleted as per scenario outline value");
+        elementInteractions.TakeScreenShot();
+        //Assertion for successful delete
+        DeleteShareSkillAssertion();
```

Comment:
so everything looks good, just move all asserts  into tests, best practice to not put them in pages!